src/main/java/ru/testtask/service/BackgroundJobService.java

```diff
@@ -0,0 +1,114 @@
+package ru.testtask.service;
+
+
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;
+import org.springframework.stereotype.Service;
+import ru.testtask.dto.BackgroundJobDTO;
+import ru.testtask.model.BackgroundJob;
+import ru.testtask.model.BackgroundJobMongo;
+import ru.testtask.model.BackgroundJobStatus;
+import ru.testtask.repo.BackgroundJobRepo;
+
+import javax.annotation.PostConstruct;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+@Service
+@Slf4j
+public class BackgroundJobService {
+    @Autowired
+    private BackgroundJobRepo backgroundJobRepo;
+
+    private final ThreadPoolTaskScheduler threadPoolTaskScheduler = new ThreadPoolTaskScheduler();
+
+    private final List<BackgroundJob> runningBackgroundJobs = new ArrayList<>();
+
+    private final int jobIterations = 20;
+
+    @PostConstruct
+    public void init(){
+        threadPoolTaskScheduler.setDaemon(true);
+        threadPoolTaskScheduler.setPoolSize(5);
+        threadPoolTaskScheduler.initialize();
+    }
+
+    public void startJob(){
+      threadPoolTaskScheduler.execute(Objects.requireNonNull(runningBackgroundJobs.get(runningBackgroundJobs.size() - 1)
```

Comment:
Why last job, not created in this thread? You just can add parameter to the method: `BackgroundJob job`